### DISCUSSION TITLE
The virtual document in browse mode is buggy when using UIA in Edge

### DISCUSSION THREAD

**Discussion by Adriani90** [nvaccess/nvda#16441] [General]
### Steps to reproduce:
1. Open NVDA
2. Open a browser using UIA (preferably Edge)
3. In NVDA advanced settings, enable UIA for chromium browsers
4. Open web pages such as 
www.google.com
www.github.com
any news websites etc.
5. Make sure you are using browse mode
6. Navigate the websites
6.1. using arrow keys
6.2. try to trigger a dialog via an action on the website and access that dialog
6.3 Use i or shift+i to navigate by list items

### Actual behavior:
Different issues happen which are not there when using browse mode with UIA disabled:
6.1.:
* Label of many Buttons and links cannot be navigated by character, only a space character along with the full label is announced
* Some of links or buttons (e.g. on Github) are not fully rendered in the virtual document, only the first letter is rendered as buutton (e.g. C for "close button") while the rest of the label is accessible only with left and right arrow and it is rendered as plain text (e.g. the remaining "lose" from the close button label). Using up and down arrow only "C" is reported instead of "Close".
* Screen layout off is not respected: sometimes links are sticked together and rendered as plain text when using up and down arrow keys, even though screen layout is turned off in NVDA browse mode settings (e.g. on google start page)
* In some cases every character of a link label is rendered on a separate line (e.g. the "skip to content" link in the header of the Github page)
* Replacements characters "0xffc) and separator characters such as breaking lines in css / html are rendered on separate lines (e.g. on Google page when navigating search results with up and down arrow keys)

6.2.:
* Sometimes o or shift+o does not find embeded objects such as dialogs
* Dialog is reported on every item when focusing dialog items with tab or shift+tab
* Buttons such as "close button" are not fully rendered as buttons in the virtual document, see 6.1 above.

6.3:
* It is not possible to navigate to list items with i or shift+i, the command seems not supported. The l quick nav command is supported though.
* Following quick nav commands are also impacted: a, m, q, u, v

### Expected behavior:
6.1: NVDA should render links and buttons in virtual document correctly, with their full label, navigable by character. Screen layout setting should be respected.
6.2. Dialogs should always be reachable with o and shift+o command, and pressing tab or shift+tab on dialog items should not report "dialog" on every item.
6.3. NVDA should support all quick nav commands that are also supported when UIA is disabled.

### NVDA logs, crash dumps and other attachments:
n/a

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-31645,bf96860d (2024.2.0.31645)

#### Windows version:
Windows 11 23 H2

#### Name and version of other software in use when reproducing the issue:
Edge Chromium 124

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If NVDA add-ons are disabled, is your problem still occurring?
No add-ons in use

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
cc: @tomaszw, @aursulis I think these problems are also affecting the HP browser.

**Comment by seanbudd** [comment]
These need to be filed as separate issues (if they don't exist already). There are a lot of known issues with UIA in chromium, its is incomplete, however filing remaining work is useful if it is broken up into distinct tasks.

**Comment by Adriani90** [comment]
In this case it would have made sense to convert it first to a discussion, but ok.I think I will not split because if we really start workink towards a solution, the virtual document needs to be taken as a whole.Von meinem iPhone gesendetAm 23.04.2024 um 01:50 schrieb Sean Budd ***@***.***>:﻿
These need to be filed as separate issues (if they don't exist already). There are a lot of known issues with UIA in chromium, its is incomplete, however filing remaining work is useful if it is broken up into distinct tasks.

**Reply by seanbudd** [reply]
I've converted it to a discussion. 
I think if we are just noting general inaccessibility with UIA Chromium - a meta issue may be appropriate.
That way the issue can note that a lot of work is needed in UIA Chromium and potentially link and group issues there.
